katsdpcontroller/product_config.py

```diff
@@ -374,6 +414,62 @@ def from_config(cls,
         )
 
 
+class NgcAntennaChannelisedVoltageStream(AntennaChannelisedVoltageStreamBase):
+    """Real antenna-channelised-voltage stream (next-gen correlator).
+
+    It currently only supports wideband.
+    """
+
+    stream_type: ClassVar[str] = 'ngc.antenna_channelised_voltage'
+    _valid_src_types: ClassVar[_ValidTypes] = {'sim.dig.raw_antenna_voltage'}
+
+    def __init__(self, name: str, src_streams: Sequence[Stream], *,
+                 n_chans: int) -> None:
+        if n_chans < 1 or (n_chans & (n_chans - 1)) != 0:
+            raise ValueError('n_chans is not a power of 2')
+        if len(src_streams) % 2 != 0:
+            raise ValueError('src_streams does not have an even number of elements')
+        first = src_streams[0]
+        assert isinstance(first, SimDigRawAntennaVoltageStream)
+        antenna_names = []
+        for src in src_streams:
+            assert isinstance(src, SimDigRawAntennaVoltageStream)
+            if src.antenna.name not in antenna_names:
+                antenna_names.append(src.antenna.name)
+            if src.band != first.band:
+                raise ValueError(f'Inconsistent bands (both {first.band} and {src.band})')
+            if src.adc_sample_rate != first.adc_sample_rate:
+                raise ValueError(
+                    'Inconsistent ADC sample rates '
+                    f'(both {first.adc_sample_rate} and {src.adc_sample_rate})')
+            if src.centre_frequency != first.centre_frequency:
+                raise ValueError(
+                    'Inconsistent centre frequencies '
+                    f'(both {first.centre_frequency} and {src.centre_frequency})')
+        super().__init__(
+            name, src_streams,
+            antennas=antenna_names,
+            band=first.band,
+            n_chans=n_chans,
+            bandwidth=first.adc_sample_rate * 0.5,
+            adc_sample_rate=first.adc_sample_rate,
+            centre_frequency=first.centre_frequency,
+            n_samples_between_spectra=2 * n_chans
+        )
+
+    @classmethod
+    def from_config(cls,
+                    options: Options,
+                    name: str,
+                    config: Mapping[str, Any],
+                    src_streams: Sequence['Stream'],
+                    sensors: Mapping[str, Any]) -> 'NgcAntennaChannelisedVoltageStream':
```

Comment:
`sensors` not used again. Is this a pattern or a convention that the argument list needs to have this in it?